minetest/minetest
# Improvements to colored shadows

A number of improvements to colored shadows:
* Fix minetest#11361 - apply vertex color when rendering colored shadow
* Remove visual artifacts in colored shadows by disabling the trilinear filter when sampling SM texture. No visual impact since filtering is done in shaders
* Make colored shadows more prominent by applying them with multiplication instead of mix().
* Avoid rendering colored shadow on surfaces that are turned away from sun.

This PR is Ready for Review.

## How to test

* Enable colored shadow
* Install [stained_glass](https://content.minetest.net/packages/andersje/stained_glass/) mod
* `/giveme stained_glass:stained_glass`
* Place a number of stained glass blocks
* Use painter from unified_dyes mod to color the blocks yellow, red, green or blue
* Observe the casted shadows, they should match the color of the blocks
